android/java/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.java

```diff
@@ -207,6 +208,62 @@ public void setGroup(String name, String[] value, EventOptions options, Middlewa
         this.client.setGroup(name, jsonValue, extra);
     }
 
+    public void groupIdentify(String groupType, String groupName, Group event) {
+        this.groupIdentify(groupType, groupName, event, null, null);
+    }
+
+    public void groupIdentify(String groupType, String groupName, Group event, EventOptions options) {
+        this.groupIdentify(groupType, groupName, event, options, null);
+    }
+
+    public void groupIdentify(String groupType, String groupName, Group event, MiddlewareExtra extra) {
+        this.groupIdentify(groupType, groupName, event, null, extra);
+    }
+
+    public void groupIdentify(String groupType, String groupName, Group event, EventOptions options, MiddlewareExtra extra) {
+        if (!this.isInitializedAndEnabled()) {
+            return;
+        }
+        this.handleEventOptions(options, null);
+        com.amplitude.api.Identify identify = new com.amplitude.api.Identify();
+        JSONObject groupProperties = this.getEventPropertiesJson(event);
+        if (groupProperties != null) {
+            Iterator<String> keys = groupProperties.keys();
+            while (keys.hasNext()) {
+                String key = keys.next();
+
+                Object value;
+                try {
+                    value = groupProperties.get(key);
+                } catch (JSONException e) {
+                    System.err.println("Can't get value for a group property '" + key + "': " + e);
+                    continue;
+                }
+
+                if (value instanceof Boolean) {
+                    identify.set(key, ((Boolean)value).booleanValue());
+                } else if (value instanceof Double) {
+                    identify.set(key, ((Double)value).doubleValue());
+                } else if (value instanceof Float) {
+                    identify.set(key, ((Float)value).floatValue());
+                } else if (value instanceof Integer) {
+                    identify.set(key, ((Integer)value).intValue());
+                } else if (value instanceof Long) {
+                    identify.set(key, ((Long)value).longValue());
+                } else if (value instanceof String) {
+                    identify.set(key, (String)value);
+                } else if (value instanceof JSONObject) {
+                    identify.set(key, (JSONObject)value);
+                } else if (value instanceof JSONArray) {
+                    identify.set(key, (JSONArray)value);
+                } else {
+                    System.err.println("Invalid type encountered for a group property '" + key + "': " + value);
```

Comment:
would `JSONObject.NULL` falls into here?

Reply:
Not sure what options is the best:
1. skip `JSONObject.NULL` values in `groupIdentify` only
2. skip `null` values in `getEventPropertiesJson` - no `JSONObject.NULL` values in any events
3. use deprecated `public Identify set(String property, Object value)` function (https://github.com/amplitude/Amplitude-Android/blob/main/src/main/java/com/amplitude/api/Identify.java#L1739) to set `JSONObject.NULL` values in `groupIdentify`

Reply:
In Kotlin we ignore `null` values - maybe we should use that behavior in all languages/runtimes
https://github.com/amplitude/ampli-examples/blob/main/android/kotlin/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.kt#L179

Reply:
We do not currently support null values in the web app. I think we can skip/ignore null values for now.
<img width="241" alt="image" src="https://user-images.githubusercontent.com/5790872/154532089-827a99b5-708e-4efd-824a-0459308eba05.png">

Reply:
If we can ignore `null` like Kotlin that is also good. In Java the object can be passed in with null I believe, so even though we don't have the types they can also pass in Null value. If we prefer to skip that for now it may be fine, as long as this is not breaking.